lib/IMAP/PreviewEnhancer.php

```diff
@@ -50,9 +61,13 @@ public function __construct(IMAPClientFactory $clientFactory,
 	 * @return Message[]
 	 */
 	public function process(Account $account, Mailbox $mailbox, array $messages): array {
-		$needAnalyze = array_reduce($messages, static function (array $carry, Message $message) {
+		$needAnalyze = array_reduce($messages, function (array $carry, Message $message) {
 			if ($message->getStructureAnalyzed()) {
 				// Nothing to do
+				if ($message->getAvatar() === null) {
+					$avatar = $this->avatarService->getAvatar($message->getFrom()->first()->getEmail(), $this->UserId);
```

Comment:
getFrom might return an empty error. will that lead to a NPE when you call `getEmail` on null?